code/game/turfs/space/space.dm

```diff
@@ -69,7 +69,7 @@
 
 /turf/space/Entered(atom/movable/A as mob|obj)
 	..()
-	if ((!(A) || src != A.loc))
+	if ((!(A) || !(src in A.locs)))

```

Comment:
This may have some performance impact, because BYOND is absolutely atrocious at searching lists. If you moved this entire check under if(destination_z), it would be fine.
ie:
```
/turf/space/Entered(atom/movable/A)
    ..()

    if(destination_z && A && (src in A.locs))
        A.x = destination_x
        etc...
```